trl/trainer/online_dpo_trainer.py

```diff
@@ -241,14 +248,36 @@ def __init__(
             self.stats["objective/scores_margin"] = []
             self.stats["objective/scores"] = []
 
-        self.generation_config = GenerationConfig(
-            max_new_tokens=args.max_new_tokens,
-            temperature=args.temperature,
-            top_k=0,
-            top_p=1.0,
-            do_sample=True,
-            use_cache=False if args.gradient_checkpointing else True,
-        )
+        if args.use_vllm:
+            if not is_vllm_available():
+                raise ImportError(
+                    "vLLM is not available and `use_vllm` is set to True. Please install vLLM with "
+                    "`pip install vllm` to use it."
+                )
+            self.generation_config = SamplingParams(
+                n=2,  # 2 generations per prompt
+                max_tokens=args.max_new_tokens,
+                temperature=args.temperature,
+                top_k=50,
+                top_p=1.0,
+                detokenize=False,  # to avoid vllm to decode (we don't need it)
+            )
+            # vLLM dynamically adjusts the size of the key-value cache based on available GPU memory at instanciation.
+            # A larger cache size improves speed, so we would expect gpu_memory_utilization=1.
+            # However, at this stage, the optimizer's weights are not yet loaded onto the GPU; they will be loaded
+            # after the first optimizer step and remain in GPU memory throughout training. So we must reserve enough
+            # space for them. Setting gpu_memory_utilization to 0.6 seems to work well in practice.
+            self.llm = LLM(model=model.name_or_path, gpu_memory_utilization=0.55, dtype=torch.float32)
```

Comment:
`dtype` and `gpu_memory_utilization`  are hardcoded but we can still make them as arg in the future.
See here why using `torch.float32` is important.